File: android/src/main/java/com/nitroinputmask/CustomMaskEngine.kt
package com.nitroinputmask

internal class CustomMaskEngine(mask: String) : MaskEngineProtocol {
  private val compiled: CompiledMask = MaskEngine.compile(mask)

  override fun apply(input: String): Pair<String, String> {
    return MaskEngine.apply(input, compiled)
  }

  override val wantsTrailingCursor: Boolean = false
  override val expandedMask: String? = compiled.expandedMask
}

File: ios/masks/CustomMaskEngine.swift
import NitroModules

struct CustomMaskEngine: MaskEngineProtocol {
  private let compiled: CompiledMask

  init(mask: String) {
    self.compiled = MaskEngine.compile(mask: mask)
  }

  func apply(input: String) -> (masked: String, raw: String) {
    return MaskEngine.apply(input: input, compiled: compiled)
  }

  var wantsTrailingCursor: Bool { false }
  var expandedMask: String? { compiled.expandedMask }
}
